app/scripts/modules/core/src/managed/constraints/ConstraintCard.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+import { IStatefulConstraint, StatefulConstraintStatus } from '../../domain';
+import { NoticeCard } from '../NoticeCard';
+
+import { isConstraintSupported, getStatefulConstraintConfig } from './constraintRegistry';
+
+const { NOT_EVALUATED, PENDING, PASS, FAIL, OVERRIDE_PASS, OVERRIDE_FAIL } = StatefulConstraintStatus;
+
+const constraintCardAppearanceByStatus = {
+  [NOT_EVALUATED]: 'neutral',
+  [PENDING]: 'info',
+  [PASS]: 'success',
+  [FAIL]: 'error',
+  [OVERRIDE_PASS]: 'success',
+  [OVERRIDE_FAIL]: 'error',
+} as const;
+
+const throwUnsupportedConstraintError = (type: string) => {
+  throw new Error(`Unsupported constraint type ${type} — did you check for constraint support before rendering?`);
+};
+
+export interface IConstraintCardProps {
+  constraint: IStatefulConstraint;
+  className?: string;
+}
+
+export const ConstraintCard = ({ constraint, className }: IConstraintCardProps) => {
+  const { type, status } = constraint;
+
+  if (!isConstraintSupported(type)) {
+    throwUnsupportedConstraintError(type);
```

Comment:
Would be nice to avoid throwing and causing the application to break. Instead we could render a 'Not supported' or 'Invalid' card (or not render anything depending on the use case) and provide a way to alert us of these failing cases through logs.

Reply:
I agree, I would prefer to see a very crude "unsupported type" card (kind of like the JSON text area for stages without a specific stage config UI), over throwing.

Reply:
I realize now I probably should've explained this better. Here's my thinking, tell me if it's reasonable or not.

At least at the moment, we're taking the approach of not showing constraints at all if they're not explicitly supported by deck. That may change later, but it seems appropriate for now (we have a variety of constraints we expect to keep 'implicit'). This `ConstraintCard` thing is an internal component for use inside the environments view code (i.e. I do not plan on folks being able to render this themselves). In order to properly enforce the contract that we should not be attempting to show unsupported constraints, I expect every parent component rendering this code to filter out things that are unsupported and handle them properly, as opposed to having `ConstraintCard` handle it by rendering `null` and maybe end up with the parent component showing a janky layout due to expecting certain sizing/etc.

So my expectation here was that this is a totally impossible case to even get into, and the purpose of handling it at all is to quickly and efficiently assist whoever is seeing it in fixing their code (i.e. during local development). If we're worried about a bug eventually making it outside of the local dev stage and crashing customer pages, I'm not opposed to switching to rendering `null` and still `console.error`-ing this exception so it gets a nice component stack trace. Though I suspect that will make a little less obvious and harder to fix pre-deployment.

Reply:
For now I updated this to `console.error` the same exception, and return `null`. We can see how that goes and if we end up wanting stronger guardrails of some kind to avoid unhandled constraints getting haphazardly rendered